pkg/devfile/adapters/docker/component/adapter.go

```diff
@@ -61,19 +61,9 @@ func (a Adapter) Push(parameters common.PushParameters) (err error) {
 	}
 	s.End(true)
 
-	// Get the supervisord volume
-	supervisordLabels := utils.GetSupervisordVolumeLabels()
-	supervisordVolumes, err := a.Client.GetVolumesByLabel(supervisordLabels)
+	a.supervisordVolumeName, err = utils.CreateAndInitSupervisordVolume(a.Client)
```

Comment:
Couple questions around always initializing the supervisord volume:
- Can we change it so that it's only on odo pushes that require a container start/restart? If the container is running and we just need to sync the code + exec some command, we shouldn't need to re-init the volume, no?
- Does running this on every odo push impact performance?

Reply:
Regd point 2, it takes only 1s when we init volume every time.

Reply:
addressed in review commit